sdk/go/pulumi/config/require.go

```diff
@@ -53,7 +53,11 @@ func RequireObject(ctx *pulumi.Context, key string, output interface{}) {
 
 func requireBool(ctx *pulumi.Context, key, use, insteadOf string) bool {
 	v := require(ctx, key, use, insteadOf)
-	return cast.ToBool(v)
+	o, err := cast.ToBoolE(v)
+	if err != nil {
+		contract.Failf("unable to parse required configuration variable '%s'; %s", key, err.Error())
```

Comment:
Huh, weird that we use `contract.Fail` in the config APIs. Might consider just changing these to raw `panic`s.